### ISSUE TITLE
[tune] TuneError raised when some trial(s) failed inside an experiment 

### ISSUE THREAD

**Issue by old-bear** [ray-project/ray#2915]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: MacOS
- **Ray installed from (source or binary)**: binary
- **Ray version**: 0.5.2
- **Python version**: 2.7
- **Exact command to reproduce**:

### Describe the problem

If some trials fail inside an experiment, `run_experiement` will throw a `TuneError`.
```
    errored_trials = []
    for trial in runner.get_trials():
        if trial.status != Trial.TERMINATED:
            errored_trials += [trial]

    if errored_trials:
        raise TuneError("Trials did not complete", errored_trials)
```
Sometimes it's an acceptable failure outside when there are lots of trials and only one of them fails. I think we can let the caller to decide whether to throw or not (caller can check failure by himself using the returned trials). If exception is still required, we can use a different exception to distinguish  from other TuneError.

**Comment by ericl** [comment]
A flag sounds good to me.

**Comment by richardliaw** [comment]
Addressed in #2961
